packages/_infra/src/edge/index.ts

```diff
@@ -0,0 +1,50 @@
+import cdk = require('@aws-cdk/core');
+import cf = require('@aws-cdk/aws-cloudfront');
+import s3 = require('@aws-cdk/aws-s3');
+import { LambdaApiKeyValidator } from './lambda.edge.api.key';
+
+/**
+ * Edge infrastructure
+ *
+ * This will setup all the edge infrastructure needed for basemaps, including:
+ * * CloudFront CDN
+ * * API Tracking Lambda@edge
+ */
+export class EdgeStack extends cdk.Stack {
+    public constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+        super(scope, id, props);
+
+        const lambda = new LambdaApiKeyValidator(this, 'LambdaEdge');
+
+        // Empty test bucket to have a origin for the edge
+        // This will eventually be the actual tile generation origin
+        const bucket = new s3.Bucket(this, 'EmptyBucket');
+        bucket.grantPublicAccess(); // Public for now...
+
+        const lambdaViewerRequest: cf.LambdaFunctionAssociation = {
+            eventType: cf.LambdaEdgeEventType.VIEWER_REQUEST,
+            lambdaFunction: lambda.version,
+        };
+
+        const originConfig: cf.SourceConfiguration = {
+            s3OriginSource: { s3BucketSource: bucket },
+            behaviors: [
+                {
+                    isDefaultBehavior: true,
+                    lambdaFunctionAssociations: [lambdaViewerRequest],
+                },
+            ],
+        };
+
+        const cloudFront = new cf.CloudFrontWebDistribution(this, 'Distribution', {
+            priceClass: cf.PriceClass.PRICE_CLASS_ALL, // Allow Sydney as a origin
+            httpVersion: cf.HttpVersion.HTTP2,
+            viewerProtocolPolicy: cf.ViewerProtocolPolicy.REDIRECT_TO_HTTPS, // Do not allow http connections
```

Comment:
I'm wondering if we still want to allow HTTP, in the context of WMTS. Some GIS client still have issue with HTTPS...

Reply:
Do you know which GIS clients they are? 

There are a lot of configuration options for TLS with Cloudfront, I generally use the TLS2018 but they will cause issues with older devices. So having a list would be super helpful

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/secure-connections-supported-viewer-protocols-ciphers.html

Reply:
Generally there are low usage client such as Mapinfo and others. In the past even QGIS has been an issue. Mostly due to people being still on older versions of software and not upgrading (either due to cost or corporate reasons).

Root causes have been client library lack of support for newer TLS standards or even that CA root certs were not correctly installed in desktop environments.

Reply:
Just did a quick survey, both google and OSM support tile servers not using `https` so I will take this out.

Openstreetmaps is interesting though, while it will respond over http, it has HSTS set so your browser will force you into https if you try and access their tiles

Reply:
Great.